roboteam_ai/include/roboteam_ai/interface_api/InterfaceSubscriber.h

```diff
@@ -6,11 +6,11 @@
 #define RTT_INTERFACESUBSCRIBER_H
 #include <atomic>
 #include "proto/NewInterface.pb.h"
+#include <QtNetwork>
 
 namespace rtt::ai::io {
 class InterfaceGateway;
 class InterfaceSubscriber {
-    friend InterfaceGateway;  /// Only the InterfaceGateway can create an InterfacePublisher
```

Comment:
Why did you remove this and made the constructor public?

Reply:
The `InterfaceSubscriber` is now holding a QUdpSocket, which means that it needs to be instantiated. Since the `InterfaceSubscriber` is wrapped in a `std::unique_ptr`, it can only be instantiated if the constructor is public